### PR TITLE
[VDO-5698] Ingest commits: switch logger namespace from uds to vdo

### PR DESCRIPTION
**Pull request by C2Redhat**
This changes contains four commits.  

The first commit is to ingest changes from Mike Snitzer,  "dm vdo logger: change from uds_ to vdo_ namespace".  
The second commit is to adapt changes from Mike's commit.  Also, update our internal codes from uds to vdo namespaces to match the kernel changes.
The third commit is to ingest change from Mike Snitzer, "dm vdo string-utils: change from uds_ to vdo_ namespace".
The last commit is to adapt the string-utils namespace change and also rename uds_fixed_sprintf to vdo_fixed_sprintf.

### CODE REVIEW

**Review by raeburn — APPROVED** [review]

File: src/c++/uds/kernelLinux/uds/uds-module.c [hunk 1]
```diff
@@ -62,9 +60,11 @@ EXPORT_SYMBOL_GPL(vdo_duplicate_string);
 EXPORT_SYMBOL_GPL(vdo_free);
 EXPORT_SYMBOL_GPL(vdo_free_funnel_queue);
 EXPORT_SYMBOL_GPL(vdo_funnel_queue_poll);
+EXPORT_SYMBOL_GPL(vdo_get_log_level);
 EXPORT_SYMBOL_GPL(vdo_get_memory_stats);
 EXPORT_SYMBOL_GPL(vdo_is_funnel_queue_empty);
 EXPORT_SYMBOL_GPL(vdo_make_funnel_queue);
+EXPORT_SYMBOL_GPL(vdo_log_backtrace);
 EXPORT_SYMBOL_GPL(vdo_perform_once);
```
Comment (lorelei-sakai) [inline src/c++/uds/kernelLinux/uds/uds-module.c:68]: vdo_log_backtrace shoud go before vdo_make_funnel_queue.

File: src/c++/uds/userLinux/uds/logger.c [hunk 2]
```diff
@@ -116,7 +116,7 @@ static void init_logger(void)
 	bool is_abs_path = false;
 	if (log_file != NULL) {
 		is_abs_path =
-			(make_abs_path(log_file, &log_file) == UDS_SUCCESS);
+			(make_abs_path(log_file, &log_file) == VDO_SUCCESS);
 		errno = 0;
```
Comment (lorelei-sakai) [inline src/c++/uds/userLinux/uds/logger.c:120]: Don't change the error code here. The right check is probably "!= 0" but it's not related to this commit at any rate.

**Review by lorelei-sakai — CHANGES_REQUESTED** [review]
Just a few small issues.

Also, in the first commit message, please remove the line about permassert renames since that isn't in this change any more.

**Review by lorelei-sakai — APPROVED** [review]
Looks good then,

**Review by lorelei-sakai — APPROVED** [review]
Looks good.

**Pull request merged**